coalib/parsing/Filter.py

```diff
@@ -0,0 +1,19 @@
+from coalib.parsing.FilterImplementation import *
+
+
+class Filter:
+    all_filters = {'language': LanguageFilter}
+
+    def __init__(self):
+        pass
+
+    def is_valid_filter(filter):
+        return filter in Filter.all_filters
+
+    @staticmethod
+    def get_filtered_bears(filter, args, log_printer):
```

Comment:
Is there any issue taking care of it ? If not, I can file it and also start working on it. IMO, it is going to be time consuming to remove dependency from `log_printer`.